### PR TITLE
InstaLite Connector

### PR DESCRIPTION
**Pull request by bhavik-m**
- [x] Build Connector
- [x] Deploy Connector - [here](https://etherscan.io/address/0x82B5bB99e2A8133d51Be2338D654D64528263100#code)
- [x] Enable Connector
- [x] Add to SDK https://github.com/Instadapp/dsa-connect/pull/147/
- [x] Add to Docs https://github.com/Instadapp/docs/pull/160

### CODE REVIEW

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * @dev 
+
+ */
+import { TokenInterface } from "../../common/interfaces.sol";
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+import { Events } from "./events.sol";
+import { Helpers } from "./helpers.sol";
+
+abstract contract Resolver is Events, DSMath, Basic, Helpers {
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
abstract contract InstaLiteConnector is Events, DSMath, Basic, Helpers {
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		instaLite.withdraw(_amt, to);
+
+		setUint(setId, _amt);
+
+		_eventName = "LogWithdraw(uint256,address,uint256,uint256)";
+		_eventParam = abi.encode(_amt, to, getId, setId);
+	}
+}
+
+contract ConnectV2InstaLite is Resolver {
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
contract ConnectV2InstaLiteVault1 is Resolver {
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+/**
+ * @title InstaLite Connector
+ * @dev 
+
+ */
+import { TokenInterface } from "../../common/interfaces.sol";
+import { DSMath } from "../../common/math.sol";
```
Comment (Vaibhav20201) [inline contracts/mainnet/connectors/instaLite/main.sol]: Are we using DSMath anywhere?

**Comment by Vaibhav20201** [comment]
<pre>
Lets structure the instalite connectors something like
mainnet -> connectors -> instalite -> v1 (This is the one you made)
                                   -> v2
                                   ......
polygon -> connectors -> instalite -> v1
                                   -> v2
                                   ......
.
.
. 
</pre>

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	 * @param amt The amount of token to be supplied.
+	 * @param to The address of the account on behalf of you want to supplied.
+	 * @param getId ID to retrieve amt.
+	 * @param setIds ID stores the amount of token deposited.
+	 */
+	function supply(
+		address vaultAddress,
+		address token,
+		uint256 amt,
+		address to,
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]: @bhavik-m @Vaibhav20201 To will be DSA address only. I.e `address(this)`

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	 * @param vaultAddress Address of vaultAddress Contract.
+	 * @param amt The amount of the token to withdraw.
+	 * @param to The address of the account on behalf of you want to withdraw.
+	 * @param getId ID to retrieve amt.
+	 * @param setIds ID stores the amount of token withdrawn.
+	 */
+	function withdraw(
+		address vaultAddress,
+		uint256 amt,
+		address to,
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]: @bhavik-m @Vaibhav20201 same over here

File: contracts/mainnet/connectors/instaLite/main.sol (hunk 4 above)
Comment (bhavik-m) [reply contracts/mainnet/connectors/instaLite/main.sol]: I have updated the codebase

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+ */
+import { TokenInterface } from "../../common/interfaces.sol";
+import { Basic } from "../../common/basic.sol";
+import { Events } from "./events.sol";
+import { IInstaLite } from "./interface.sol";
+
+abstract contract InstaLiteConnector is Events, Basic {
+	/**
+	 * @dev Supply
+	 * @notice Supply eth/weth/stEth tokens into instalite.
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol:17]:
```suggestion
	 * @notice Supply ETH/wETH/stETH tokens into instalite.
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	 * @param vaultAddress Address of instaLite Contract.
+	 * @param token The address of token to be supplied.
+	 * @param amt The amount of token to be supplied.
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
	 * @dev Supply ETH/ERC20
	 * @notice Supply a token into Instalite.
	 * @param vaultAddress Address of instaLite Contract.
	 * @param token The address of the token to be supplied. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
	 * @param amt The amount of token to be supplied. (For max: `uint256(-1)`)
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+	 * @param getId ID to retrieve amt.
+	 * @param setIds ID stores the amount of token deposited.
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
	 * @param setIds array of IDs to store the amount of tokens deposited.
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	/**
+	 * @dev Withdraw
+	 * @notice Withdraw eth/stEth tokens from instalite contract.
+	 * @param vaultAddress Address of vaultAddress Contract.
+	 * @param amt The amount of the token to withdraw.
+	 * @param getId ID to retrieve amt.
+	 * @param setIds ID stores the amount of token withdrawn.
+	 */
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
	 * @dev Withdraw ETH/ERC20
	 * @notice Withdraw deposited tokens from Instalite.
	 * @param vaultAddress Address of vaultAddress Contract.
	 * @param amt The amount of the token to withdraw.
	 * @param getId ID to retrieve amt.
	 * @param setIds array of IDs to stores the amount of tokens withdrawn.
	 */
```

File: contracts/mainnet/connectors/instaLite/main.sol [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+		setUint(setIds[0], _amt);
+		setUint(setIds[1], vTokenAmt);
+
+		_eventName = "LogWithdraw(address,uint256,uint256,uint256,uint256[])";
+		_eventParam = abi.encode(vaultAddress, _amt, vTokenAmt, getId, setIds);
+	}
+}
+
+contract ConnectV2InstaLiteVault1 is InstaLiteConnector {
+	string public constant name = "instaLite-v1";
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite/main.sol]:
```suggestion
	string public constant name = "InstaLite-v1";
```

**Review by thrilok209 — DISMISSED** [review]

**Review by Vaibhav20201 — APPROVED** [review]

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**